[OptimistLabyrinth] 🚀 1단계 - 지하철역 인수 테스트 작성

OptimistLabyrinth opened the pull request:
RestAssured 관련해서는 검색해도 찾을 수 있는 정보가 너무 제한적이라서
약간 어려움을 겪고 있습니다.

할 수 있는 선에서 최대한 해봤는데
피드백 주시면 수정해보겠습니다.

Commit:
feat: 인수 테스트 추가

지하철역 리스트 조회
지하철역 삭제

wheejuni commented on src/test/java/nextstep/subway/station/StationAcceptanceTest.java:
```diff
@@ -101,6 +102,30 @@ void createStationWithDuplicateName() {
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .when().post("/stations")
+                .then().log().all();
+        params.put("name", "역삼역");
+        RestAssured.given().log().all()
+                .body(params)
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .when().post("/stations")
+                .then().log().all();
```
하나의 독립적인 테스트 메소드로 분리해서 작성되어야 할 것 같군요.

OptimistLabyrinth replied:
수정했습니다.

wheejuni commented on src/test/java/nextstep/subway/station/StationAcceptanceTest.java:
```diff
@@ -101,6 +102,30 @@ void createStationWithDuplicateName() {
+                        .extract();
+
+        // then
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.OK.value());
```
단순 로그를 찍고 `statusCode` 를 보는 것이 요구사항이 아닌 것으로 저는 이해하고 있습니다.
`JsonPath` DSL을 사용해 리스폰스 바디가 실제 요구 사항과 합치하는지를 봐야 합니다. [이것](https://www.javadoc.io/doc/io.rest-assured/json-path/3.0.0/io/restassured/path/json/JsonPath.html) 을 참조해 보시면 좋겠네요.

OptimistLabyrinth replied:
수정 완료했습니다.

wheejuni reviewed (changes requested):
생소한 과제 진행하시느라 고생하셨습니다. 👍 
약간의 의견 드리겠습니다~

Commit:
feat: 인수테스트 개선

1. 반복적으로 등장하는 로직 별도 메소드로 추출

StationAcceptanceTest.createStation

2. 응답코드 뿐만 아니라 HTTP Response Body 에 들어있는 데이터 내용도 검증

3. final 키워드 추가

불변성 보장

OptimistLabyrinth commented:
인수테스트에서 필요해 보이는 사항 추가했습니다.

[1]
반복적으로 등장하는 로직은 별도 메서드로 분리했습니다.
StationAcceptanceTest 클래스 createStation 메서드가 예시입니다.

[2]
응답코드를 확인하는 동작 외에도
HTTP Response Body 에 들어있는 데이터 내용을 통해서
코드가 정상적으로 실행되었는지 확인하고 있습니다.

[3]
불변성 객체에는 final 키워드를 추가했습니다.

OptimistLabyrinth commented:
추가적으로 피드백 주시면 거기에 맞춰서 수정해볼께요.

wheejuni commented on src/test/java/nextstep/subway/station/StationAcceptanceTest.java:
```diff
@@ -30,86 +32,161 @@ public void setUp() {
+                    RestAssured.given().log().all()
+                            .when().get("/stations")
+                            .then().log().all()
+                            .extract().jsonPath().getList("name", String.class);
```
👍 좋습니다! 스스로 잘 찾아내셨네요.

wheejuni reviewed (approved):
👍 지금 단계에선 문제 없어 보입니다. 수고하셨습니다.
